executor/union_scan.go

```diff
@@ -276,8 +275,18 @@ func (us *UnionScanExec) compare(a, b []types.Datum) (int, error) {
 func (us *UnionScanExec) buildAndSortAddedRows() error {
 	us.addedRows = make([][]types.Datum, 0, len(us.dirty.addedRows))
 	mutableRow := chunk.MutRowFromTypes(us.retTypes())
-	for h, data := range us.dirty.addedRows {
+	t, found := GetInfoSchema(us.ctx).TableByID(us.dirty.tid)
+	if !found {
+		// t is got from a snapshot InfoSchema, so it should be found, this branch should not happen.
+		return errors.Errorf("table not found (tid: %d)", us.dirty.tid)
```

Comment:
Could we add the current infoschema version to the error message?

Reply:
ok

Reply:
PTAL @zimulala